website/docs/internals/resource-addressing.html.markdown

```diff
@@ -72,3 +81,33 @@ aws_instance.web
 ```
 
 Refers to all four "web" instances.
+
+### for_each Example
+
+Given a Terraform config that includes:
+
+```hcl
+resource "aws_instance" "web" {
+  # ...
+  for_each = {
+    "terraform": "value1",
+    "resource":  "value2",
+    "indexing":  "value3",
+    "example":   "value4",
+  }
+}
+```
+
+An address like this:
+
+```
+aws_instance.web["example"]
+```
+
+Refers to only the "example" instance in the config, and an address like this:
+
+```
+aws_instance.web[*]
```

Comment:
This smells like it needs explication earlier! If I'm understanding it right, this means two things are missing from the "Resource index" section above:

- For resources created with for_each, the index can be either an explicit index key, or else the wildcard key `*`. 
- You can't omit the index to refer to all instances of a resource configured with for_each; you have to use `[*]`.

y/n?

Reply:
Sorry this is a little bit of an oversight on my part. Either omitting `[*]` or using `[*]` should work depending on the end goals. It's likely clearest to just remove this little section here.